samples/snippets/src/main/java/com/example/spanner/ChangeStreamSample.java

```diff
@@ -0,0 +1,626 @@
+/*
+ * Copyright 2021 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+// [START spanner_change_stream]
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ErrorCode;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.Struct;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import com.google.spanner.admin.database.v1.CreateDatabaseMetadata;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/**
+ * Sample code for Cloud Spanner Change Stream. It demonstrates how to:
+ * 1. Set up the Cloud Spanner client.
+ * 2. Create a database with a table that's watched by a change stream.
+ * 3. Insert rows into the table.
+ * 4. Execute change stream initial query to get change stream partition tokens.
+ * 5. Execute a change stream partition query to get data change records of the inserted rows.
+ */
+public class ChangeStreamSample {
+  public static void main(String... args) {
+    if (args.length != 2 && args.length != 3) {
+      System.err.println("Usage: ChangeStreamSample "
+          + "<project_id> "
+          + "<instance_id> "
+          + "<database_id> (optional)");
```

Comment:
To simplify the sample I think that the database id should not be optional

Reply:
Should we even keep database_id as an argument then? If we keep it, then that means user will need to have a database with Singers and ChangeStreamAll, otherwise we can't insert and query change stream. I lean forward to just remove database_id, and we always create a temp database, what do you think?

Reply:
Sorry, I was thinking of keeping the database id, but creating the tables during the sample initialization (before creating the change streams).